internal/aws/cwlogs/pusher.go

```diff
@@ -54,10 +55,33 @@ func NewEvent(timestampMs int64, message string) *Event {
 	return event
 }
 
-// Uniquely identify a cloudwatch logs stream
+// Uniquely identify a cloudwatch logs stream. Any changes to this struct will require updates to Hash
 type StreamKey struct {
 	LogGroupName  string
 	LogStreamName string
+	Entity        *cloudwatchlogs.Entity
+}
+
+// Custom hash function for StreamKey. Necessary to uniquely identify with Entity.
+func (sk *StreamKey) Hash() string {
+	var attributes, keyAttributes string
+	if sk.Entity != nil {
+		if sk.Entity.Attributes != nil {
+			attributes = mapToString(sk.Entity.Attributes)
+		}
+		if sk.Entity.KeyAttributes != nil {
+			keyAttributes = mapToString(sk.Entity.KeyAttributes)
+		}
+	}
+
+	data := fmt.Sprintf(
```

Comment:
One concern I have with the hash function is that we need to make sure we update it if the struct changes.  We should have a unit test that validates the number of fields on StreamKey and Entity do not change.

Reply:
addressed by adding a unit test that validates number of fields, and the contents/types of the fields